pkg/protoconv/resources/resources_test.go

```diff
@@ -141,6 +141,21 @@ func TestCronJobPopulateSpec(t *testing.T) {
 	assert.Equal(t, deploymentWrap.Containers[0].Name, "container2")
 }
 
+func TestNewDeploymentFromStaticResourcePopulatesPodLabels(t *testing.T) {
+	deployment := &appsV1.Deployment{
+		Spec: appsV1.DeploymentSpec{
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": "nginx"},
+				},
+				Spec: v1.PodSpec{Containers: []v1.Container{{Name: "nginx"}}},
+			},
+		},
+	}
+	d, _ := NewDeploymentFromStaticResource(deployment, "Deployment", "", "")
```

Comment:
We are in test, so checking the error returned from a function is recommended and may save us from potential bugs in the future. Let's maybe add an assertion that the second param is nil.